okhttp-tests/src/test/java/okhttp3/CallTest.java

```diff
@@ -2867,11 +2867,60 @@ private void cancelDuringConnect(String scheme) throws Exception {
     Response response = client.newCall(request).execute();
     assertEquals("response body", response.body().string());
 
+    RecordedRequest connect1 = server.takeRequest();
+    assertNull(connect1.getHeader("Proxy-Authorization"));
+
+    RecordedRequest connect2 = server.takeRequest();
+    assertEquals("password", connect2.getHeader("Proxy-Authorization"));
+  }
+
+  /** Confirm that we can send authentication information without being prompted first. */
+  @Test public void preemptiveProxyAuthentication() throws Exception {
```

Comment:
Might be worth adding a test showing a failed premptive proxy auth followed up by reactive attempt.

Reply:
Good call. Done!